### PR TITLE
`<BadgeSelectItem/>`- adding a uni driver

### PR DESCRIPTION
**Pull request by TalZa1810**
### 🔦 Summary
This PR includes:
- Align the component name to be `BadgeSelectItem` instead of the multiple names that we previously had.
- Creating a uni driver for `<BadgeSelectItem/>`.
- Adding tests for the uni driver.
- Rename folder and file name `BadgeSelectItemBuilder` ==> `BadgeSelectItem` 

Refactoring `BadgeSelectItem` story page will be in a dedicated PR ([Jira](https://jira.wixpress.com/browse/DSM-878))

### ✅ Checklist

- [ ] 👨‍💻 API change is approved by the UI Infra Developers 
- [ ] 👨‍🎨 UX change is approved by the Design System UX 
- 📚 Change is documented
  - [ ] Story
  - [x] API description
  - [ ] Cheatsheet
  - [ ] Other (explain)
- 🔬 Change is tested
  - [x] Component
  - [x] Visual test

### CODE REVIEW

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  You can find it under one of the following URLs:
  * https://preview.wix-style-react.com/pr_6124

**Comment by TalZa1810** [comment]
Hi @argshook , 
We want to align this builder with all our exiting builders that we have within the library. This component had 3 different names: `BadgeOption` , `BadgeSelectItem` and `BadgeSelectOption`. 
I think that I’ve aligned everything. The only thing which I need to finish with is exporting the uni driver. I’m not sure how to export this at the `definitions.js` file. I’ve tried to add [`unidriverPath` prop](https://github.com/wix/wix-style-react/blob/master/.wuf/testkits/definitions.js#L11) but it did not work as expected since it is listed under directory name `BadgeSelectItemBuilder`. The file name is `BadgeSelectItemBuilder` as well.
  
Should I rename the file names and the directory? I don’t want to break existing users and I’m not sure what’s the proper way to export the uni driver for this unique use case.

File: src/BadgeSelectItem/docs/README.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 > An option builder for the `<DropdownLayout/>` component and its consumers.
 
 ```js
-import { badgeSelectItemBuilder } from 'wix-style-react/BadgeSelectItemBuilder';
+import { badgeSelectItemBuilder } from 'wix-style-react';
```
Comment (argshook) [inline src/BadgeSelectItem/docs/README.md:6]:
```suggestion
import { BadgeSelectItem } from 'wix-style-react';
```

should this builder still be advertised for users? Maybe they should use `BadgeSelectItem` instead?

File: src/BadgeSelectItem/test/BadgeSelectItem.spec.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  const requiredProps = {
+    id: 0,
+    skin: SKINS.success,
+    text: 'text',
+  };
+
+  afterEach(cleanup);
+
+  it('should render', async () => {
+    const { driver } = render(<BadgeSelectItem {...requiredProps} />);
```
Comment (argshook) [inline src/BadgeSelectItem/test/BadgeSelectItem.spec.js:20]: this test is not needed, it's covered by the `testkit-smoke.test.js`

File: testkit/component-definitions.js [hunk 3]
```diff
@@ -36,6 +36,14 @@ export default {
     },
   },
 
+  BadgeSelectItem: {
+    props: {
+      id: 0,
+      skin: 'general',
+      text: 'Badge 1',
```
Comment (argshook) [inline testkit/component-definitions.js:43]: i see only `text` prop is required, so it should not be needed to define other props here

File: src/BadgeSelectItem/docs/README.md (hunk 1 above)
Comment (TalZa1810) [reply src/BadgeSelectItem/docs/README.md:6]: Up until now only the builder was exposed. Now, we would like to export both the builder (which can be used for `DropDownLayout` ) and the component (which is exposed in this PR).
We should create a new story page anyways so I thought doing this there.

File: testkit/component-definitions.js (hunk 3 above)
Comment (TalZa1810) [reply testkit/component-definitions.js:42]: skin is also a required prop. just id is relevant for the `DropDownLayout` and not the `BadgeSelectItem` itself.

**Comment by TalZa1810** [comment]
No visual changes for this one. Therefore merging.

**Pull request merged**